### DISCUSSION TITLE
AMQP 1.0 PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'QueueQ' in vhost 'dev': received 'classic' but current is 'quorum'

### DISCUSSION THREAD

**Discussion by KrysiaMasinska** [rabbitmq/rabbitmq-server#8394] [Other] [answered]
### Describe the bug

I'm using plugin [rabbitmq-amqp1.0](https://github.com/rabbitmq/rabbitmq-amqp1.0) . When I consuming messages from the queue type quorum I'm getting the below error: PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'QueueQ' in vhost 'dev': received 'classic' but current is 'quorum'

### Reproduction steps

1. ReceiverLink _receiver = new ReceiverLink(_manager.Session, "dev", "QueueQ");
2. Getting error: {utf8,<<"PRECONDITION_FAILED - inequivalent arg 'durable' for queue 'QueueQ' in vhost 'dev': received 'false' but current is 'true'">>},
3. So, I change my reciverLink like that:
ReceiverLink _receiver = new ReceiverLink(_manager.Session, "dev", new Source() { Address = "QueueQ", Durable = 2 }, null);
4. And now I'm getting other error:
PRECONDITION_FAILED - inequivalent arg 'x-queue-type' for queue 'QueueQ' in vhost 'dev': received 'classic' but current is 'quorum'

### Expected behavior

The messages should be consumed

### Additional context

How should I pass "x-queue-type" in ReceiverLink settings?
I'm using library https://github.com/Azure/amqpnetlite

**Comment by KrysiaMasinska** [comment]
server: 3.11.6

**Comment by mkuratczyk** [comment]
This is not a RabbitMQ issue - it's a mismatch between your server-side configuration (where the queue exists and is of type `quorum`) and your client application trying to use a `classic` queue of the same name. Most likely your environment is being transitioned from classic mirrored queues to quorum queues and your app doesn't know about this.

Given you have the option to adjust the queue arguments in your code, I'd recommend specifying `x-queue-type=quorum`. This should solve the problem. For those who can't easily do that, we recently added an option to ignore this mismatch:
https://github.com/rabbitmq/rabbitmq-server/blob/main/release-notes/3.11.16.md#enhancements

**Comment by michaelklishin** [comment, marked as answer]
You can learn more in [Declaration and Property Equivalence](https://rabbitmq.com/queues.html#property-equivalence).

Note that in 3.11, virtual hosts can be configured to use a default queue type different from `classic`, see `rabbitmqctl help update_vhost_metadata`.

I am not suggesting that you change the default, but it can be the factor that has made the queue you are trying to use a quorum queue.
